presto-native-execution/presto_cpp/main/PrestoExchangeSource.cpp

```diff
@@ -350,6 +360,9 @@ void PrestoExchangeSource::abortResults() {
 
 void PrestoExchangeSource::close() {
   closed_.store(true);
+  if (promise_.valid() && !promise_.isFulfilled()) {
```

Comment:
exchange close is called by operator close and is there any potential race condition between async triggered request and the driver close? Do we need source lock to protecdt the update or access to request promise? Thanks!